packages/flame/lib/src/components/position_component.dart

```diff
@@ -164,14 +164,33 @@ class PositionComponent extends Component {
     return Vector2(width * scale.x.abs(), height * scale.y.abs());
   }
 
+  /// The resulting size after all the ancestors and the components own scale
+  /// has been applied.
+  Vector2 get absoluteScaledSize {
+    final absoluteScale = this.absoluteScale;
+    return Vector2(
+      width * absoluteScale.x.abs(),
+      height * absoluteScale.y.abs(),
+    );
+  }
+
   /// The resulting angle after all the ancestors and the components own angle
   /// has been applied.
   double get absoluteAngle {
-    return ancestors()
-        .whereType<PositionComponent>()
+    // TODO(spydon): take scale into consideration
```

Comment:
Will this be done in a follow up?

Reply:
Yes, this can't be handled with our current nested system, we'd have to know in each step what transformations that has been done before it and in which order.